ext/nokogiri/extconf.rb

```diff
@@ -560,7 +560,8 @@ def install
       *(libiconv_recipe ? "--with-iconv=#{libiconv_recipe.path}" : iconv_configure_flags),
       "--with-c14n",
       "--with-debug",
-      "--with-threads"
+      "--with-threads",
+      *(darwin? ? "RANLIB=/usr/bin/ranlib AR=/usr/bin/ar" : "")
```

Comment:
This will generate a Makefile which includes a line like `RANLIB = /usr/bin/ranlib AR=/usr/bin/ar`.
It should be `*(darwin? ? %w[RANLIB=/usr/bin/ranlib AR=/usr/bin/ar] : "")

Reply:
I don't understand why you say it generates a `Makefile`. It doesn't. Configure options are given to the `configure` script

Reply:
`configure` generates `Makefile` with given arguments.
You can check it as follows:

```
git clone https://github.com/sparklemotion/nokogiri.git
curl https://github.com/sparklemotion/nokogiri/pull/1813.patch | git apply
bundle
bundle exec rake
grep -H RANLIB tmp/*/nokogiri/*/tmp/*/ports/libxml2/*/libxml2-*/Makefile
```

It will show something like
```
tmp/x86_64-darwin16/nokogiri/2.5.3/tmp/x86_64-apple-darwin16.7.0/ports/libxml2/2.9.8/libxml2-2.9.8/Makefile:RANLIB = /usr/bin/ranlib
```

Reply:
Sure. 

I meant that I didn't think that `*(darwin? ? "RANLIB=/usr/bin/ranlib AR=/usr/bin/ar" : "")` was a problem. Do you mean it's given as a single positional argument to `configure` instead of 2?

Anyways, I updated the PR

Reply:
Yeah, `*(darwin? ? "RANLIB=/usr/bin/ranlib AR=/usr/bin/ar" : "")` will be given as sinle argument and handled like `ENV["RANLIB"] = "/usr/bin/ranlib\ AR=/usr/bin/ar"`. It's not expected behavior.

Reply:
Thanks, I have to admit I didn't check myself since just forcing `RANLIB` worked for me 😉